src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -235,129 +132,115 @@ const vizFn = function (
+    this.forceSim.updateRelationships(this.graph)
+  }
 
-    const nodes = graph.nodes()
-    const relationships = graph.relationships()
+  private updateRelationships() {
+    const relationships = this.graph.relationships()
+    this.geometry.onGraphChange(this.graph, {
+      updateNodes: false,
+      updateRelationships: true
+    })
 
-    const relationshipGroups = container
+    const relationshipGroups = this.container
       .select('g.layer.relationships')
-      .selectAll<BaseType, Relationship>('g.relationship')
+      .selectAll<SVGGElement, Relationship>('g.relationship')
       .data(relationships, d => d.id)
       .join('g')
       .attr('class', 'relationship')
-      .on('mousedown', onRelationshipClick)
-      .on('mouseover', onRelMouseOver)
-      .on('mouseout', onRelMouseOut)
+      .call(relationshipEventHandlers, this.trigger)
       .classed('selected', relationship => relationship.selected)
 
-    for (const renderer of vizRenderers.relationship) {
-      relationshipGroups.call(renderer.onGraphChange, viz)
-    }
+    relationshipRenderer.forEach(renderer =>
+      relationshipGroups.call(renderer.onGraphChange, this)
+    )
 
-    function dragHandler(simulation: Simulation<VizNode, Relationship>) {
-      let initialPos: [number, number]
-      let restartedSim = false
-      const tolerance = 25
-
-      const dist = (a: [number, number], b: [number, number]) =>
-        Math.pow(a[0] - b[0], 2) + Math.pow(a[1] - b[1], 2)
-
-      function dragstarted(event: D3DragEvent<SVGGElement, VizNode, any>) {
-        initialPos = [event.x, event.y]
-        restartedSim = false
-      }
-
-      function dragged(
-        event: D3DragEvent<SVGGElement, VizNode, any>,
-        node: VizNode
-      ) {
-        // This is to prevent clicks/double clicks from restarting the simulation
-        if (dist(initialPos, [event.x, event.y]) > tolerance && !restartedSim) {
-          // Set alphaTarget to a value higher than alphaMin so the simulation
-          // isn't stopped while nodes are being dragged.
-          simulation.alphaTarget(0.09).alpha(1).restart()
-          restartedSim = true
-        }
+    this.forceSim.updateRelationships(this.graph)
+  }
 
-        node.hoverFixed = false
-        node.fx = event.x
-        node.fy = event.y
-      }
+  private zoomClick(isZoomingIn: any) {
+    this.draw = true
+    this.isZoomClick = true
+    const limitsReached = { zoomInLimit: false, zoomOutLimit: false }
 
-      function dragended(_event: D3DragEvent<SVGGElement, VizNode, any>) {
-        if (restartedSim) {
-          // Reset alphaTarget so the simulation cools down and stops.
-          simulation.alphaTarget(0)
-        }
-      }
+    this.zoomBehavior.scaleBy(this.baseGroup, isZoomingIn ? 1.3 : 0.7)
 
-      return d3Drag<SVGGElement, VizNode>()
-        .on('start', dragstarted)
-        .on('drag', dragged)
-        .on('end', dragended)
-    }
+    return limitsReached
```

Comment:
Should we remove this return value if we don't know if we hit the limits anymore?

Reply:
Same question here.